docs/manual/src/udl/callback_interfaces.md

```diff
@@ -168,3 +153,71 @@ Take care to avoid this by following guidelines:
    references through intermediate objects.
 3. If you need to break the first 2 guidelines, then take steps to manually break the cycles to avoid memory leaks.
    Alternatively, ensure that the number of objects that can ever be created is bounded below some acceptable level.
+
+## Scheduling tasks using the `TaskQueue` callback interface
```

Comment:
This is orthogonal to user-defined callback interfaces, but I think it could be really useful.  I'm thinking we should implement them at more-or-less the same time.

Reply:
Not to say we shouldn't, but from what I read here it might be possible to define and implement this outside of UniFFI.
If so, what benefits does it have to make that part of uniffi directly?

Reply:
The issue is that it needs a couple of things not supported in UniFFI:
  - Storing/calling an arbitrary Rust closure
  - Generics (it returns the closure return type)

Both of those could be nice features to support in general and once we do, then we don't need to special case this.  However,  I don't see that happening anytime soon.

Reply:
don't mean to bikeshed too much, but I wonder if something like `TaskExecutor` is a better name here?